src/macros.jl

```diff
@@ -643,7 +530,7 @@ function _process_func_expr(_error::Function, raw_expr)
         func_expr = esc(raw_expr)
         is_anon = true
     else 
-        _error("Unrecognized syntax.")
+        error_fn("Unrecognized syntax.")
```

Comment:
This PR would be a lot simpler if you split the `error_fn` stuff into a separate PR.